### PR TITLE
Another pre-commit to check Pylint score

### PR DESCRIPTION
**Pull request by wjduand**
Run vagrant provision to install needed package.
Use another version of pre-commit so that we can set score limit.
You can edit score limit in .pylintrc file to test the functionality.
It should run automatically when you commit.

### CODE REVIEW

**Comment by nisargthakkar** [comment]
Hi @wjduand. I tested this and it works. I had to make a couple of changes, though. We need a way to set the content of `.git/hooks/pre-commit`.

How about creating a file in our directory called `pre-commit`, and in the `Vagrantfile`, just before the chmod, we can copy the file in `.git/hooks` and then run the `chmod` command that you already have.

What do you think?

File: requirements.txt [hunk 1]
```diff
@@ -4,10 +4,13 @@ Flask-API==1.1
...
 
 # Testing
 nose==1.3.7
 pinocchio==0.4.2
 factory-boy==2.12.0
 coverage==4.5.4
-pylint>=2.4.1
\ No newline at end of file
+pylint>=2.4.1
+pre-commit==1.18.3
```
Comment (nisargthakkar) [inline requirements.txt]: Do we still need `pre-commit`? I think we can remove this since we are manually setting the data in the `.git/hooks/pre-commit`.

Comment (wjduand) [reply requirements.txt]: You are right I guess it should have be git-pylint-commit-hook in the testing section

**Comment by wjduand** [comment]
> Hi @wjduand. I tested this and it works. I had to make a couple of changes, though. We need a way to set the content of `.git/hooks/pre-commit`.
> 
> How about creating a file in our directory called `pre-commit`, and in the `Vagrantfile`, just before the chmod, we can copy the file in `.git/hooks` and then run the `chmod` command that you already have.
> 
> What do you think?

Yes there should be a pre-commit file with content:
 #!/usr/bin/env bash
git-pylint-commit-hook
But I'm not sure why it's not included in this commit. And why can't we just add the file instead of copying it?

**Comment by nisargthakkar** [comment]
That's because the `.git` folder is ignored by git when committing. 

Copying a custom file is somewhat similar to what `pre-commit install` would have done. They put some content into the `.git/hooks/pre-commit` file so that the checks can be run

**Comment by wjduand** [comment]
> That's because the `.git` folder is ignored by git when committing.
> 
> Copying a custom file is somewhat similar to what `pre-commit install` would have done. They put some content into the `.git/hooks/pre-commit` file so that the checks can be run

That makes sense. I can change that and make a commit.

**Comment by wjduand** [comment]
Hi @nisargthakkar could you check this now?

File: requirements.txt [hunk 2]
```diff
@@ -4,10 +4,12 @@ Flask-API==1.1
 Flask-SQLAlchemy==2.4.1
 psycopg2-binary==2.8.3
 pymysql==0.9.3
+git-pylint-commit-hook
```
Comment (nisargthakkar) [inline requirements.txt]: Sorry, I just noticed. I think we can remove this line as well.

File: Vagrantfile [hunk 3]
```diff
@@ -62,6 +63,8 @@ Vagrant.configure(2) do |config|
     # Install app dependencies
     cd /vagrant
     pip3 install -r requirements.txt
+	cp pre-commit .git/hooks/pre-commit
```
Comment (nisargthakkar) [inline Vagrantfile]: Can you also fix the indentation here?

File: requirements.txt (hunk 2 above)
Comment (nisargthakkar) [reply requirements.txt]: this is because we have it in the Testing area.

Comment (wjduand) [reply requirements.txt]: Do we? Where is it?

Comment (nisargthakkar) [reply requirements.txt]: Down in the requirements.txt only where you've added it. (My phrasing might be wrong. I didn't mean testing code. Only testing section of requirements.txt)

Comment (wjduand) [reply requirements.txt]: Got it.

**Review by nisargthakkar — APPROVED** [review]
LGTM

**Comment by sflechas** [comment]
@wjduand is the main comment of this thread updated with the latest instructions to get this change? I will download and provision my instance to get these changes.  If not, could you please update them?  Thank you.

**Comment by wjduand** [comment]
> @wjduand is the main comment of this thread updated with the latest instructions to get this change? I will download and provision my instance to get these changes. If not, could you please update them? Thank you.

Yes everything is updated.

**Pull request merged**